snuba/consumers/schemas.py

```diff
@@ -25,11 +26,20 @@ def get_schema(topic: Topic) -> Optional[Mapping[str, Any]]:
         return None
 
 
-_cache: MutableMapping[Topic, JsonCodec[Any]] = {}
+class NoopCodec(JsonCodec[T]):
+    def __init__(self):
+        pass
 
+    def validate(self, data: T) -> None:
+        pass
 
-def get_json_codec(topic: Topic) -> JsonCodec[Any]:
+
+_NOOP_CODEC = NoopCodec()
```

Comment:
Do we really need the user to define this NoopCodec? Isn't that the default behavior for JsonCodec if a schema isn't provided?

Reply:
The schema argument in codecs is now non-nullable. I thought it made sense to change that. We can however revert it. I'm just not sure if we want to keep the consumer entirely clueless about whether there is a schema at all

Reply:
I guess my question is, what is the default return value for get-codec if there is no schema? is it specifically JsonCodec?

Reply:
Makes sense. We would need to put both a JsonCodec and MsgPackCodec in there. Not sure if it is worth it.

Reply:
What about if we bring over this from the arroyo implementation. So we don't need to define the noop codec in many places: https://github.com/getsentry/sentry-kafka-schemas/pull/123

Reply:
merged and released

Reply:
thanks